### ISSUE TITLE
[cookbook] Adding http handler for handling backend auth in the background

### ISSUE THREAD

**Issue by ivansieder** [vuejs/vuejs.org#1819]
For some applications (which make use of JWTs), we are using fetch and custom Vue instance methods to handle fetch requests and re-authentication in the background, so in case the access token has expired, the application will automatically retry fetching a new access token with the refresh token, if for a request it gets a 401. If that request was successful, it will retry fetching the resource one more time.

There is actually already an awesome cookbook about custom instance properties (https://vuejs.org/v2/cookbook/adding-instance-properties.html), but this one would specifically talk about a $http handler which also takes care of authentication in the background automatically.

Would this be something of interes? If so, I'd love to write something up and create a PR for it.

**Comment by sdras** [comment]
Hi Ivan! Yes I think this would make a great cookbook entry. It might make sense to link from one to the other so that people who are interested can see a particular use case in action. Thanks! I look forward to your PR.

**Comment by thestamp** [comment]
I for one would love this, with an example of how to handle expired sessions (ex: login popover)

**Comment by ivansieder** [comment]
Yes @thestamp. That’s what i was planning. With the example of using JWTs. The request would automatically try to refetch a new access token and then retry the request in case of an initial 401 for example. Same could be for your case. Opening the login form for a 401 in the response. Stay tuned!

**Comment by ivansieder** [comment]
Almost same day, different year. Gonna go and make this one now, as it's been open on my bucket list for waaay too long. @sdras as far as I can tell, cookbooks should always be isolated (no other resources required), is that correct? What about linking to the "Adding Instance Properties" for this tutorial, as it actually uses the same technique?
